controller/share_manager_controller.go

```diff
@@ -567,8 +567,10 @@ func (c *ShareManagerController) cleanupShareManagerPod(sm *longhorn.ShareManage
 // controls transitions to running, error
 func (c *ShareManagerController) syncShareManagerPod(sm *longhorn.ShareManager) (err error) {
 	defer func() {
-		if sm.Status.State == longhorn.ShareManagerStateStopping || sm.Status.State == longhorn.ShareManagerStateStopped ||
+		if sm.Status.State == longhorn.ShareManagerStateStopping ||
+			sm.Status.State == longhorn.ShareManagerStateStopped ||
 			sm.Status.State == longhorn.ShareManagerStateError {
+			sm.Status.IP = ""
```

Comment:
Why requiring `sm.Status.IP = ""`? Don't the other 3 state conditions cover all cases to clean up the SM pod?

Reply:
It is not necessary now. Removed. Thank you.